federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -998,6 +1008,9 @@ func (s *ServiceController) processServiceUpdate(cachedService *cachedService, s
 	defer cachedService.rwlock.Unlock()
 
 	if service.DeletionTimestamp != nil {
+		// Ensure that we propagate the DeletionTimestamp
+		cachedService.lastState = service
+		s.serviceCache.set(key, cachedService)
```

Comment:
Should we be doing this only if it hasnt already been deleted from serviceCache?